src/core/column/column.service.js

```diff
@@ -1,10 +1,28 @@
+import {isFunction} from '../services/utility';
+import AppError from '../infrastructure/error';
+
 export function map(columns) {
 	return columns.reduce((memo, column) => {
 		memo[column.key] = column;
 		return memo;
 	}, {});
 }
 
+export function getValue(column) {
+	return isFunction(column.value)
+		? row => column.value(row)
+		: row => row[column.key];
+}
+
+export function find(columns, key) {
+	for (let i = 0, length = columns.length; i < length; i++) {
+		if (columns[i].key == key) {
+			return columns[i];
+		}
+	}
+	throw new AppError('Not Found', `column ${key} is not found`);
```

Comment:
first arg should be column.service, Column should be in upper case, $key decorated with double quotes "
also when call find doesn't expect exception can you return null, and throw outside?